### PR TITLE
Added postgres branch to build trigger

### PR DESCRIPTION
**Pull request by mofterdinger**
- added postgres branch to build trigger
- updated actions to latest versions

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -5,20 +5,30 @@ name: Java CI with Maven
...
     steps:
-    - uses: actions/checkout@v2
+    - name: Checkout
+      uses: actions/checkout@v4
     - name: Set up JDK
-      uses: actions/setup-java@v1
+      uses: actions/setup-java@v4
       with:
         java-version: 17
+        distribution: 'temurin'
```
Comment (beckermarc) [inline .github/workflows/maven.yml:28]: Could we specify SAP Machine here?

Comment (mofterdinger) [reply .github/workflows/maven.yml:28]: no, not available: https://github.com/actions/setup-java?tab=readme-ov-file#supported-distributions
I checked it already, I would also prefer to use SAPMachine

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -5,20 +5,30 @@ name: Java CI with Maven
...
+    - name: Setup Maven
+      uses: stCarolas/setup-maven@v4.5
+      with:
+        maven-version: 3.9.6
```
Comment (beckermarc) [inline .github/workflows/maven.yml:32]: With which version did we run earlier? `stCarolas` sounds a little inofficial. Hopefully this is trustworthy ;)

Comment (mofterdinger) [reply .github/workflows/maven.yml:32]: > With which version did we run earlier?

To be honest I don't know, I guess it uses the maven version that is pre-installed in the container of the github runner.

> stCarolas sounds a little inofficial. Hopefully this is trustworthy ;)

it's listed here: https://github.com/marketplace/actions/setup-maven-action
I'm using it already for a couple of month in a private project without any problmes.

Comment (beckermarc) [reply .github/workflows/maven.yml:32]: Okay, I am wondering if the additional Maven setup build times are worth it. I guess this involves some download and so on. If the version we are using at the moment also works fine :)

Comment (mofterdinger) [reply .github/workflows/maven.yml:32]: ok, will remove it again. It was just some kind of preparation for the next major release when we bump the min. maven version. Then we can add it if required.

**Review by beckermarc — APPROVED** [review]

**Pull request merged**